### ISSUE TITLE
How to set extra parameters for encoding?

### ISSUE THREAD

**Issue by thesby** [UKPLab/sentence-transformers#611]
When using model `LongformerModel` of [transformers](https://huggingface.co/transformers/model_doc/longformer.html#longformermodel), it accepts `attention_mask `  and `global_attention_mask `. But how to set them in sentence_transformers when using `encode`?

**Comment by nreimers** [comment]
I never worked with the LongformerModel, so I don't know which parameters have to be passed to it and how to get it.

For the other transformer models, the tokenizer produces the necessary inputs (input_ids, attention_mask), which are passed from the tokenizer to the transformer model.

**Comment by thesby** [comment]
Thank you. I want to set `global_attention_mask` by myself. But I don't find how to set it by `SentenceTransformer`. It's a tensor to set attention points of long sequence.

**Comment by nreimers** [comment]
You have to create a models. class like models.Transformer. There, the method 'get_sentence_features' defines the features for every sentence. You would need to change it so that the right global attention mask is used there.

I think it would require quite some changes.

**Comment by thesby** [comment]
Thank you. I got it.
